src/components/NewCommentForm.tsx

```diff
@@ -1,103 +1,227 @@
-import React from 'react';
-
-export const NewCommentForm: React.FC = () => {
-  return (
-    <form data-cy="NewCommentForm">
-      <div className="field" data-cy="NameField">
-        <label className="label" htmlFor="comment-author-name">
-          Author Name
-        </label>
-
-        <div className="control has-icons-left has-icons-right">
-          <input
-            type="text"
-            name="name"
-            id="comment-author-name"
-            placeholder="Name Surname"
-            className="input is-danger"
-          />
-
-          <span className="icon is-small is-left">
-            <i className="fas fa-user" />
-          </span>
-
-          <span
-            className="icon is-small is-right has-text-danger"
-            data-cy="ErrorIcon"
-          >
-            <i className="fas fa-exclamation-triangle" />
-          </span>
-        </div>
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { createComment } from '../utils/commentService';
+import { Comment } from '../types/Comment';
+import { usePosts } from './PostsProvider';
+
+type Props = {
+  setPostComments: (comm: Comment[]) => void;
+  postComments: Comment[];
+  setErrorMessage: (err: string) => void;
+};
+
+export const NewCommentForm: React.FC<Props>
+  = ({ setErrorMessage, postComments, setPostComments }) => {
+    const [name, setName] = useState('');
+    const [hasNameError, setHasNameError] = useState(false);
+    const [email, setEmail] = useState('');
+    const [hasEmailError, setHasEmailError] = useState(false);
+    const [comment, setComment] = useState('');
+    const [hasCommentError, setHasCommentError] = useState(false);
+    const [loading, setLoading] = useState(false);
+
+    const { selectedPost } = usePosts();
+
+    const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setName(event.target.value);
```

Comment:
Try to use useRef instead. It prevents ur component from rerender in every letter u typed
https://react.dev/reference/react/useRef